.github/workflows/build_and_test.yml

```diff
@@ -948,7 +948,8 @@ jobs:
 
   deploy:
     needs: [ check-permissions, promote-images, tag, build-and-test-locally, trigger-custom-extensions-build-and-wait, push-to-acr-dev, push-to-acr-prod ]
-    if: github.ref_name == 'main' || github.ref_name == 'release' || github.ref_name == 'release-proxy'
+    # `!failure() && !cancelled()` is required because the workflow depends on the job that can be skipped: `push-to-acr-dev` and `push-to-acr-prod`
```

Comment:
I don't understand this comment. How are the `failure` and `cancelled` related to skipping?

Reply:
Well.. because there are 4 possible statuses: `success`, `failure`, `cancelled`, `skipped`, but there are only 3 functions to check the status: `success()`, `failure()` and `cancelled()`, (there's also `always()` that runs the job always ([GitHub docs](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#status-check-functions)))

So if you need to run the job on `success` and `skipped`, it means **not** to run it on `failed` and `cancelled`.